tiledb/sm/filesystem/vfs.cc

```diff
@@ -75,6 +75,12 @@ VFS::VFS() {
   STATS_FUNC_VOID_OUT(vfs_constructor);
 }
 
+VFS::~VFS() {
+#ifdef HAVE_S3
+  s3_.disconnect();
```

Comment:
Is there a better place for this than the destructor, e.g. a de-init? Disconnect can return an error status.

Reply:
At some point, we'll just have to log an error and move on. I've added a VFS::terminate() routine that invokes this and propagates the returned status. I decided to have our top-level C-API and storage manager just log an error if this new `terminate` fails.

We can discuss later if we want to make a `terminate` routine part of our top-level, user API.